psi4/driver/schema_wrapper.py

```diff
@@ -586,6 +586,16 @@ def run_json_qcschema(json_data, clean, json_serialization, keep_wfn=False):
             )):
                 json_data["extras"]["qcvars"][k] = _serial_translation(v, json=json_serialization)
 
+    # Add in handling of matrix arguments which need to be obtained by a
+    # a function call.
+    if "tIJAB" in json_data["extras"]["qcvars"]:
+        json_data["extras"]["qcvars"]["tIJAB"] = wfn.get_amplitudes("tIJAB").to_array()
+    if "tIA" in json_data["extras"]["qcvars"]:
+        json_data["extras"]["qcvars"]["tIA"] = wfn.get_amplitudes("tIA").to_array()
+    if "Da" in json_data["extras"]["qcvars"]:
+        json_data["extras"]["qcvars"]["Da"] = wfn.Da().to_array()
```

Comment:
any way you could use the density from the wavefunction schema instead of resaving it here? or is it wrong basis and cumbersome reordering?

Reply:
are correlated densities properly updated in wavefunction schema? I know the `wfn` density generally only updates it from the scf if there was an `OEProp` or `properties` call, but at least that's expected. just a case of the devil you know vs the devil you don't :)